Userland/Libraries/LibWeb/HTML/HTMLScriptElement.cpp

```diff
@@ -49,236 +50,264 @@ void HTMLScriptElement::begin_delaying_document_load_event(DOM::Document& docume
+        || (!has_type_attribute && !has_language_attribute)) {
         script_block_type = "text/javascript";
-    } else if (has_type) {
-        // Otherwise, if the script element has a type attribute, let the script block's type string for this script element be the value of that attribute.
-        script_block_type = attribute(HTML::AttributeNames::type);
-    } else if (!attribute(HTML::AttributeNames::language).is_empty()) {
-        // Otherwise, the element has a non-empty language attribute; let the script block's type string for this script element be the concatenation of the string "text/" followed by the value of the language attribute.
+    }
+    // Otherwise, if el has a type attribute,
+    else if (has_type_attribute) {
+        // then let the script block's type string be the value of that attribute with leading and trailing ASCII whitespace stripped.
+        script_block_type = attribute(HTML::AttributeNames::type).trim_whitespace();
+    }
+    // Otherwise, el has a non-empty language attribute;
+    else if (!attribute(HTML::AttributeNames::language).is_empty()) {
+        // let the script block's type string be the concatenation of "text/" and the value of el's language attribute.
         script_block_type = String::formatted("text/{}", attribute(HTML::AttributeNames::language));
     }
 
-    // Determine the script's type as follows:
+    // 9. If the script block's type string is a JavaScript MIME type essence match,
     if (MimeSniff::is_javascript_mime_type_essence_match(script_block_type.trim(Infra::ASCII_WHITESPACE))) {
-        // - If the script block's type string with leading and trailing ASCII whitespace stripped is a JavaScript MIME type essence match, the script's type is "classic".
+        // then set el's type to "classic".
         m_script_type = ScriptType::Classic;
-    } else if (script_block_type.equals_ignoring_case("module"sv)) {
-        // - If the script block's type string is an ASCII case-insensitive match for the string "module", the script's type is "module".
+    }
+    // 10. Otherwise, if the script block's type string is an ASCII case-insensitive match for the string "module",
+    else if (script_block_type.equals_ignoring_case("module"sv)) {
+        // then set el's type to "module".
         m_script_type = ScriptType::Module;
-    } else {
-        // - If neither of the above conditions are true, then return. No script is executed.
+    }
+    // 11. Otherwise, if the script block's type string is an ASCII case-insensitive match for the string "importmap",
+    else if (script_block_type.equals_ignoring_case("importmap"sv)) {
+        // then set el's type to "importmap".
+        m_script_type = ScriptType::ImportMap;
+    }
+    // 12. Otherwise, return. (No script is executed, and el's type is left as null.)
+    else {
         dbgln("HTMLScriptElement: Refusing to run script because the type '{}' is not recognized.", script_block_type);
```

Comment:
worth a `VERIFY(m_script_type == ScriptType::Null)` here? Although saying that, we don't have an "Invalid" state for that enum :), which is against da shpec

> A [script](https://html.spec.whatwg.org/multipage/scripting.html#the-script-element) element has a type, which is either null, "classic", "module", or "importmap", initially null. It is determined when the element is [prepared](https://html.spec.whatwg.org/multipage/scripting.html#prepare-the-script-element), based on the [type](https://html.spec.whatwg.org/multipage/scripting.html#attr-script-type) attribute of the element at that time.

Reply:
Great point! Will add `ScriptType::Null`